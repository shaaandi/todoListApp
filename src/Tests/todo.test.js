import React from 'react';
import {shallow} from 'enzyme';
import Todo from '../components/jsfiles/todo';



it('NavBar Testing', () => {
    let todo = {
                title : 'Drink Milk',
                state : false,
                steps : [],
                flag : 'white',
                viewSideBar : () => {},
                id : 'fgdgd32wersd'
    }
    let wrapper = shallow(<Todo {...todo}/>);
    expect(wrapper).toMatchSnapshot();
})