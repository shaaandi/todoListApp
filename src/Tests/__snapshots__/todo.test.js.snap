// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Testing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Todo
    flag="white"
    id="fgdgd32wersd"
    state={false}
    steps={Array []}
    title="Drink Milk"
    viewSideBar={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="stateButton"
          onClick={[Function]}
        >
          <div
            className="todoState"
          >
            <FaCircleNotch
              size={22}
            />
          </div>
        </button>,
        <div
          className="todoTitle"
          onClick={[Function]}
        >
          Drink Milk
        </div>,
        <button
          className="stateButton"
          onClick={[Function]}
        >
          <FaWindowClose
            color="red"
          />
        </button>,
      ],
      "className": "todo",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="todoState"
          >
            <FaCircleNotch
              size={22}
            />
          </div>,
          "className": "stateButton",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FaCircleNotch
              size={22}
            />,
            "className": "todoState",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": 22,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Drink Milk",
          "className": "todoTitle",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Drink Milk",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FaWindowClose
            color="red"
          />,
          "className": "stateButton",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "red",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="stateButton"
            onClick={[Function]}
          >
            <div
              className="todoState"
            >
              <FaCircleNotch
                size={22}
              />
            </div>
          </button>,
          <div
            className="todoTitle"
            onClick={[Function]}
          >
            Drink Milk
          </div>,
          <button
            className="stateButton"
            onClick={[Function]}
          >
            <FaWindowClose
              color="red"
            />
          </button>,
        ],
        "className": "todo",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="todoState"
            >
              <FaCircleNotch
                size={22}
              />
            </div>,
            "className": "stateButton",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FaCircleNotch
                size={22}
              />,
              "className": "todoState",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "size": 22,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Drink Milk",
            "className": "todoTitle",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Drink Milk",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FaWindowClose
              color="red"
            />,
            "className": "stateButton",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": "red",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
